packages/block-editor/src/components/block-mobile-toolbar/index.native.js

```diff
@@ -19,24 +19,31 @@ import styles from './style.scss';
 import BlockMover from '../block-mover';
 import { BlockSettingsButton } from '../block-settings';
 
-const BlockMobileToolbar = ( { clientId, onDelete, order } ) => (
+const BlockMobileToolbar = ( {
+	clientId,
+	onDelete,
+	order,
+	disallowRemoveBlock,
+} ) => (
 	<View style={ styles.toolbar }>
 		<BlockMover clientIds={ [ clientId ] } />
 
 		<View style={ styles.spacer } />
 
 		<BlockSettingsButton.Slot />
 
-		<ToolbarButton
-			title={ sprintf(
-				/* translators: accessibility text. %s: current block position (number). */
-				__( 'Remove block at row %s' ),
-				order + 1
-			) }
-			onClick={ onDelete }
-			icon={ trash }
-			extraProps={ { hint: __( 'Double tap to remove the block' ) } }
-		/>
+		{ ! disallowRemoveBlock && (
+			<ToolbarButton
+				title={ sprintf(
+					/* translators: accessibility text. %s: current block position (number). */
+					__( 'Remove block at row %s' ),
+					order + 1
+				) }
+				onClick={ onDelete }
+				icon={ trash }
+				extraProps={ { hint: __( 'Double tap to remove the block' ) } }
+			/>
+		) }
```

Comment:
I have concerns about bringing this because we're going to revert this after (https://github.com/WordPress/gutenberg/pull/20502) gets merged